### ISSUE TITLE
Geolocation "timeout"

### ISSUE THREAD

**Issue by pasqualepalmaccio** [onury/geolocator#54]
Hi,
when i try "geolocator.locate " function returns often "timeout" . First time works perfectly but if i reload function i get timeout. 
Also on your site if you click on locate button you will get timeout of geolocator. How can i solve this? 

Best regards

**Comment by onury** [comment]
Hi. Yes, this is really strange. This seems to happen only on **Chrome**. Is that the case for you too? Did you check it on Firefox or others?

I also found out that if you wait for at least 1 minute, between calls, it does not timeout; otherwise it will timeout everytime (after first) until 1 minute is passed. 

This is not related with Geolocator.js, it seems. Pls try this page https://browserleaks.com/geo and see if it happens there too using Chrome.

**Comment by silverbux** [comment]
hi @onury is the 1 minute limit applies per user? i mean one user will be able to use the "geolocator.locate " per minute?

thanks in advance

**Comment by onury** [comment]
@silverbux hi. This is either a Chrome bug or is done intentionally. 
But I couldn't find any official acknowledgement about this yet. 

`geolocator.locate()` internally invokes the HTML5 Geolocation API.
In my tests, I found that Chrome browser seems to limit Geolocation API call to 1 call per limit, per page. Any more calls than that, on the same page, will timeout.

Can you also confirm whether this is the case for you?
Pls also test https://browserleaks.com/geo multiple times to see if it happens on that page too, using Chrome and maybe other browsers.

Thanks.

**Comment by pasqualepalmaccio** [comment]
Hi,
i confirm that code on https://browserleaks.com/geo is working fine and i
get my position after few seconds. Your method geolocator.locate actually
not working correctly and i always get timeout if timeout parameter is set
lower than 30000



2018-04-21 22:34 GMT+02:00 Onur Yıldırım <notifications@github.com>:

**Comment by pasqualepalmaccio** [comment]
Yes I tested code in latest Chrome version and latest Safari version and I
obtain the same problem. With html5 geolocator it works well. I set
parameters with timeout:1e4 and accurancy off.

Il dom 22 apr 2018, 04:23 Onur Yıldırım <notifications@github.com> ha
scritto:

**Comment by mmolimpa** [comment]
Gblk toolkit

**Comment by isocroft** [comment]
Any progress details on this ?

**Comment by ivanoats** [comment]
I am seeing this issue on Firefox 65.0.1 on Mac OS 10.14.3

I have checked browserleaks.com/geo and it works fine on that site in Firefox.

**Comment by mucill** [comment]
Hi,
I've still got the same issue in Chrome/FF. At the first time running, it was fine. 
After that, No Response or Timeout. My friend said it must be run in https, but still the same. Even I set the timeout as 60000. Any help will appreciated. Thank you.
